src/Frames/Joins.hs

```diff
@@ -0,0 +1,191 @@
+{-# LANGUAGE ConstraintKinds, DataKinds, FlexibleContexts, FlexibleInstances,
+             KindSignatures, MultiParamTypeClasses, PolyKinds,
+             ScopedTypeVariables, TypeFamilies, TypeOperators,
+             UndecidableInstances, TemplateHaskell, QuasiQuotes,
+             Rank2Types #-}
+module Frames.Joins where
+import Data.Discrimination
+import Data.Functor.Contravariant
+import Data.Foldable as F
+import Frames
+import Frames.Melt (RDeleteAll)
+import Frames.InCore (RecVec)
+import Data.Vinyl.TypeLevel
+import Data.Vinyl
+import Data.Vinyl.Functor
+import Data.Functor.Contravariant.Divisible
+import qualified Data.Text as T
+
+dropCols :: 
```

Comment:
There are several undocumented helpers defined here. If we're sure they're only of interest to this module, we should add an explicit export list to the module and not include them. If they might be useful outside of this module, they should be documented. To clarify which definitions we expect to be most useful, we could either move the helpers into a `Frames.Joins.Internal` module, or we could re-export the main API definitions from the `Frames` module itself.

Reply:
I think `dropCols` could probably be of general use, it's essentially anti-select. The rest probably not. I will add an explicit exports list. Potentially `dropCols` could be added to exploration along-side select. Not attached to the name `dropCols`.

Reply:
Added explicit export to fix this